src/client/components/input/drag-and-drop.tsx

```diff
@@ -28,22 +28,24 @@ const {useState, useCallback} = React;
 * The Achievement interface, defining the properties of an achievement.
 * @typedef
 * @property {string} name - The name of the achievement.
-* @property {string} src - The source URL of the achievement's badge image.
-* @property {string} id - The ID of the achievement.
+* @property {string} badgeUrl - The source URL of the achievement's badge image.
+* @property {number} id - The ID of the achievement.
 */
 type Achievement = {
 	name: string;
-	src: string;
-	id: string;
+	badgeUrl: string;
+	id: number;
```

Comment:
We do set this to `null` in handleClick, I'm surprised Typescript doesn't scream about it.

The type should probably explicitly allow for this:
```suggestion
	id: number | null;
```

Reply:
Fixed